v2/contribute/decisions/thirdparty/0002-handling-static-and-dynamic-provider-list.mdx

```diff
@@ -0,0 +1,83 @@
+---
+id: "0002"
+title: Handling statically and dynamically configured provider list
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Handling statically and dynamically configured provider list
+
+<DecisionHeader status="proposed" lastUpdate="2022-12-08" created="2022-12-08" deciders={["rishabhpoddar", "sattvikc"]} proposedBy={["sattvikc"]} />
+
+## Context and Problem Statement
+
+Since we now allow users to dynamically add thirdparty provider configurations into the core, and also the user might have configured a static list of thirdparty providers in the code, this causes an ambiguity on what the final set of providers list should be, for the SDK to use.
+
+How do we come up with a consolidated list of providers?
+
+## Considered Options
+
+* Union of the list from core and the static config
+* Prioritize list from the core if not empty, else static list
+
+
+## Decision Outcome
+
+Chosen option: **Prioritize list from the core if not empty, else static list**, because
+
+* Gives complete control on the provider list for each tenant
+* Allows declaration in the static list for overrides, even if not used for all the tenants
+
+## Pros and Cons of the Options
+
+### Union of the list from core and the static config
+
+<ArgumentCon> Cannot hide a provider for a specific tenant if statically declared </ArgumentCon>
+
+### Prioritize list from the core if not empty, else static list
+
```

Comment:
Create separate ADR: The user will only give providerinput in static list and no need to go google.init etc... And now if the user gives id as google, we will just use the in built google provider.